server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java

```diff
@@ -361,6 +361,21 @@ private String login(Model model, Principal principal, List<String> excludedProm
             returnLoginPrompts = false;
         }
 
+        if (jsonResponse && request.getRequestURI().endsWith("/info")) {
+            List<Map<String, String>> infoIdentityProviders = new java.util.ArrayList<>();
+            providerProvisioning.retrieveAll(true, IdentityZoneHolder.get().getId())
+                    .forEach(provider -> {
```

Comment:
I think we have to filter , because SAML and OAUTH use a property to determine if visible or not

Reply:
not only acitve, but only provider with either
* showSamlLink
* showLinkText